### PR TITLE
Update react links

### PR DESCRIPTION
**Pull request by sasoria**
## Changes

- Updates React links to the new react documentation site (minor content fix). 
- Part of https://github.com/withastro/docs/issues/2988

## Testing

This has been tested with `pnpm run test`, but no additional tests have been added as this is a minor content fix.

## Docs
No documentation needed.

### CODE REVIEW

**Comment by sarah11918** [comment]
> There's one last link here: [examples/framework-react/README.md](https://github.com/sasoria/astro/blob/01bd16925d8a9151e2ed5ab6da0ee892dd938ec1/examples/framework-react/README.md?plain=1#L11)

@TheOtterlord, can you make a separate PR for that? Then we can merge this one and not wait for someone to add a new file to this PR!  :raised_hands:

**Review by sarah11918 — APPROVED** [review]
Approving the link updates to the new react docs! Thanks @sasoria! :rocket:

Will want someone else to approve this PR because of the changes to the `.ts` files, which update some logic around links in error messages. Maybe @Princesseuh can confirm these?

**Review by Yan-Thomas — APPROVED** [review]
Nice work @sasoria, LGTM! 

Pointing out that we might need an changeset for this for changing both integration and astro core code?

**Comment by sasoria** [comment]
Thanks for the feedback! I've added a changeset.

File: packages/astro/src/runtime/server/jsx.ts [hunk 1]
```diff
@@ -288,7 +288,7 @@ function filteredConsoleError(msg: any, ...rest: any[]) {
 		// the following error, which we can safely filter out:
 		const isKnownReactHookError =
 			msg.includes('Warning: Invalid hook call.') &&
-			msg.includes('https://reactjs.org/link/invalid-hook-call');
+			msg.includes('https://react.dev/warnings/invalid-hook-call-warning');
```
Comment (bluwy) [inline packages/astro/src/runtime/server/jsx.ts:291]: I don't think React has updated their error message links yet, so we should probably revert this: https://github.com/search?q=repo%3Afacebook%2Freact%20invalid-hook-call&type=code

File: packages/integrations/preact/src/server.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ function filteredConsoleError(msg: string, ...rest: any[]) {
 		// the following error, which we can safely filter out:
 		const isKnownReactHookError =
 			msg.includes('Warning: Invalid hook call.') &&
-			msg.includes('https://reactjs.org/link/invalid-hook-call');
+			msg.includes('https://react.dev/warnings/invalid-hook-call-warning');
```
Comment (bluwy) [inline packages/integrations/preact/src/server.ts:121]: Here too

**Review by bluwy — CHANGES_REQUESTED** [review]

**Comment by sasoria** [comment]
Good catch, I have reverted the changes.

**Review by bluwy — APPROVED** [review]
Thanks!

**Pull request merged**